homeassistant/helpers/config_validation.py

```diff
@@ -290,6 +291,45 @@ def url(value: Any) -> str:
     raise vol.Invalid('invalid url')
 
 
+def ordered_dict(value_validator, key_validator=match_all):
+    """Validate an ordered dict validator that maintains ordering.
+
+    value_validator will be applied to each value of the dictionary.
+    key_validator (optional) will be applied to each key of the dictionary.
+    """
+    def validator(value):
+        """Validate ordered dict."""
+        config = OrderedDict()
+        errors = []
+        for key, val in value.items():
+            try:
+                v_key = key_validator(key)
+            except ValueError:
+                errors.append(vol.ValueInvalid('is not a valid key', [key]))
+                continue
+            except vol.Invalid as ex:
+                errors.append(ex)
+                continue
+
+            try:
+                val = value_validator(val)
+            except ValueError:
+                errors.append(vol.ValueInvalid('is not a valid value', [key]))
```

Comment:
No, as it is the path to the value. So actually the 'key' one is not completely correct I guess. But with the patch from @kellerza that doesn't matter anymore so I'll stick with his work.